From b21f8e08308474d29de2f504f3f44d0071f48fb9 Mon Sep 17 00:00:00 2001
From: "Steven Rostedt (VMware)" <rostedt@goodmis.org>
Date: Fri, 7 Sep 2018 15:49:16 -0400
Subject: [PATCH 0433/2345] Revert "softirq: keep the 'softirq pending' check
 RT-only"

This reverts commit 5536f5491a2e098ba34995662dfc0e82d66d65c9.

Sebastian says that the stable commits:

 0a0e0829f9901 ("nohz: Fix missing tick reprogram when interrupting an inline softirq")
 80d20d35af1ed ("nohz: Fix local_timer_softirq_pending()")

Make this commit obsolete, thus it should be reverted.

Link: http://lkml.kernel.org/r/20180906193028.jvyp2dj3rto4ax7s@linutronix.de

Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>
---
 kernel/softirq.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/kernel/softirq.c b/kernel/softirq.c
index e3f89bc..583c9ec 100644
--- a/kernel/softirq.c
+++ b/kernel/softirq.c
@@ -842,13 +842,8 @@ static inline void tick_irq_exit(void)
 	int cpu = smp_processor_id();
 
 	/* Make sure that timer wheel updates are propagated */
-#ifdef CONFIG_PREEMPT_RT_BASE
 	if ((idle_cpu(cpu) || tick_nohz_full_cpu(cpu)) &&
-	    !need_resched() && !local_softirq_pending())
-#else
-	if ((idle_cpu(cpu) && !need_resched()) || tick_nohz_full_cpu(cpu))
-#endif
-	{
+	    !need_resched() && !local_softirq_pending()) {
 		if (!in_irq())
 			tick_nohz_irq_exit();
 	}
-- 
2.7.4

